### PR TITLE
Agents protocol updates

### PR DESCRIPTION
**Pull request by EngHabu**
## Tracking issue
_https://github.com/flyteorg/flyte/issues/<number>_

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

### CODE REVIEW

File: flyteidl/protos/flyteidl/admin/agent.proto [hunk 1]
```diff
@@ -11,7 +11,6 @@ import "flyteidl/core/metrics.proto";
 import "google/protobuf/duration.proto";
 import "google/protobuf/timestamp.proto";
 
-
 // The state of the execution is used to control its visibility in the UI/CLI.
 enum State {
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/admin/agent.proto:15]: why is it still in admin/agent.proto. I am ok, but just seems odd

File: flyteidl/protos/flyteidl/admin/agent.proto [hunk 2]
```diff
@@ -21,6 +20,12 @@ enum State {
   SUCCEEDED = 4;
 }
 
+// TaskOverrides is a set of overrides that can be applied to a task.
+message TaskOverrides {
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/admin/agent.proto:24]: did you not want to use this - https://github.com/flyteorg/flyte/blob/bcb7abef1e3f578511d558bd01e8fed8924b536e/flyteidl/protos/flyteidl/core/workflow.proto#L279 just to keep in separate and name clean? (I agree)

File: flyteidl/protos/flyteidl/service/agent.proto [hunk 3]
```diff
@@ -8,22 +8,53 @@ import "flyteidl/admin/agent.proto";
...
+  // ExecuteTaskSync streams the create request and inputs to the agent service and streams the outputs back.
+  rpc ExecuteTaskSync (stream flyteidl.admin.ExecuteTaskSyncRequest) returns (stream flyteidl.admin.ExecuteTaskSyncResponse){
+    option (google.api.http) = {
+      post: "/api/v1/agent/task/stream"
+      body: "*"
+    };
+  };
+
+  // CreateTask sends a task create request to the agent service.
+  rpc CreateTask (stream flyteidl.admin.CreateTaskRequest) returns (flyteidl.admin.CreateTaskResponse){
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/service/agent.proto:20]: why should this be a stream? isnt that potentially dangerous?

File: flyteidl/protos/flyteidl/admin/agent.proto [hunk 4]
```diff
@@ -35,46 +40,79 @@ message TaskExecutionMetadata {
...
+  TaskOverrides overrides = 10;
 }
 
 // Represents a request structure to create task.
 message CreateTaskRequest {
-  // The inputs required to start the execution. All required inputs must be
-  // included in this map. If not required and not provided, defaults apply.
-  // +optional
-  core.LiteralMap inputs = 1;
+  oneof part {
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/admin/agent.proto:51]: this feels odd and does not seem to have a clear understanding of what will be sent. 
Inputs and header should always be sent i feel.

IMO, CreateTask should remain sync, we can add a ExecuteSyncStream that sounds like a better option if we need stream

File: flyteidl/protos/flyteidl/admin/agent.proto [hunk 5]
```diff
@@ -35,46 +40,79 @@ message TaskExecutionMetadata {
...
+  }
+}
+
+message CreateRequestHeader {
   // Template of the task that encapsulates all the metadata of the task.
-  core.TaskTemplate template = 2;
+  core.TaskTemplate template = 1;
   // Prefix for where task output data will be written. (e.g. s3://my-bucket/randomstring)
-  string output_prefix = 3;
+  string output_prefix = 2;
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/admin/agent.proto:64]: checkpoint?

File: flyteidl/protos/flyteidl/service/agent.proto [hunk 6]
```diff
@@ -8,22 +8,53 @@ import "flyteidl/admin/agent.proto";
...
+    option (google.api.http) = {
+      post: "/api/v1/agent/task"
+      body: "*"
+    };
+  };
+
   // Get job status.
-  rpc GetTask (flyteidl.admin.GetTaskRequest) returns (flyteidl.admin.GetTaskResponse){};
+  rpc GetTask (flyteidl.admin.GetTaskRequest) returns (stream flyteidl.admin.GetTaskResponse){
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/service/agent.proto:28]: this stream also seems odd? 
Maybe this is the problem with async vs sync. 
Async being a stream is really odd?

File: flyteidl/protos/flyteidl/service/agent.proto [hunk 7]
```diff
@@ -8,22 +8,53 @@ import "flyteidl/admin/agent.proto";
 
 // AsyncAgentService defines an RPC Service that allows propeller to send the request to the agent server.
 service AsyncAgentService {
-  // Send a task create request to the agent server.
-  rpc CreateTask (flyteidl.admin.CreateTaskRequest) returns (flyteidl.admin.CreateTaskResponse){};
+  // ExecuteTaskSync streams the create request and inputs to the agent service and streams the outputs back.
+  rpc ExecuteTaskSync (stream flyteidl.admin.ExecuteTaskSyncRequest) returns (stream flyteidl.admin.ExecuteTaskSyncResponse){
```
Comment (kumare3) [inline flyteidl/protos/flyteidl/service/agent.proto:12]: i like this stream

File: flyteidl/protos/flyteidl/admin/agent.proto (hunk 1 above)
Comment (pingsutw) [reply flyteidl/protos/flyteidl/admin/agent.proto:16]: we already deprecated it

**Comment by EngHabu** [comment]
closing in favor of https://github.com/flyteorg/flyte/pull/4874

**Pull request closed without merging**